clientconn.go

```diff
@@ -532,9 +532,10 @@ func (cc *ClientConn) handleSubConnStateChange(sc balancer.SubConn, s connectivi
 // Caller needs to make sure len(addrs) > 0.
 func (cc *ClientConn) newAddrConn(addrs []resolver.Address) (*addrConn, error) {
 	ac := &addrConn{
-		cc:    cc,
-		addrs: addrs,
-		dopts: cc.dopts,
+		cc:                  cc,
+		addrs:               addrs,
+		dopts:               cc.dopts,
+		successfulHandshake: true, // make the first nextAddr() call _not_ move addrIdx up by 1
```

Comment:
Nit for clarity: "not move up addrIdx by 1" -> "reset addrIdx"?  (Is that correct?)

Reply:
No, it is as mentioned. Basically nextAddr will skip incrementing counters if it saw a successful handshake (because onPrefaceReceipt resets counters in that case). By setting it to true here, we make sure that the first iteration of the resetTransport loop does not increment counters (we want to start at 0, not 1).

I'm not sure if there's a nicer way to do this... do you want me to look into simplifying this a bit? It's a bit arcane.